contracts/FlowIDTableStaking.cdc

```diff
@@ -77,6 +77,13 @@ pub contract FlowIDTableStaking {
 
     /// The minimum amount of tokens that each node type has to stake
     /// in order to be considered valid
+    /// Keys:
+    /// 0 - Delegators
```

Comment:
I am not a big fan of combining the delegator minimum stake with the per-role stake minimums. I think we should prioritize conceptual precision over the convenience of re-using an existing storage slot.

The staking contract has the business logic concepts:
- a node role represents the set of responsibilities and capabilities associated with a node running in the network (collection, consensus, access, verification, execution)
- a staking mechanism is a way to stake tokens and receive rewards (registration, delegation)

I can register a node with a particular role. Then someone can delegate tokens to that node. I cannot register a node with role `delegator`. They are conceptually distinct and should be differentiated in the business logic.

In addition, several comments are now incorrect or imprecise with this change.

> /// Checks if the amount of tokens is greater than the minimum staking requirement for the specified role

> /// The minimum amount of tokens that each node type has to stake in order to be considered valid

Delegators are not a node type or role.

I am being a bit pedantic here, but this is a core system contract. It is important that it is as easy as possible to read and reason about, both for maintainers and community members. Conflating delegation minimums with node role minimums introduces a fairly small additional cognitive overhead to reasoning about the contract ("node roles _but also_ delegator minimums"), but the _but also_ moments add up to making bugs more difficult to spot, and the behaviour less comprehensible to community members. 

Concretely, I would suggest to store the delegator minimum stake as a separate config field with distinct documentation, setters, and getters.

Reply:
Thanks for the feedback! That is a great point. I do really like the idea of keeping it a part of the shared field with the node minimums though because it saves some lines of code and streamlines the contract in other small ways, but I understand the potential confusion. The field is just called `minimumStakeRequired`, so it could include delegators if we wanted to expand the definition of the term `role` and fix the documentation. I'm a little torn. I'll see if I can get another opinion.

Reply:
>  The field is just called minimumStakeRequired, so it could include delegators if we wanted to expand the definition of the term role and fix the documentation. I'm a little torn. I'll see if I can get another opinion.

I think this is very good idea.

Reply:
I agree with @jordanschalm here.  Readability is important and some more lines of code here is worth it to be correct.

Reply:
Not sure how to resolve this. It looks like there is fairly equal support for both paths and I still am on the side of doing it how it is already done here.